test/changeStatus.test.ts

```diff
@@ -0,0 +1,107 @@
+import { Request, Response } from 'express';
+import adminController from '../src/controllers/adminController';
+import User from '../src/database/models/userModel';
+import Mailer from '../src/utils/sendEmail';
+import disableUserTemplate from '../src/utils/emailTemplates/disableUserTemplate';
+
+
+chai.use(sinonChai);
+
+describe('adminController', () => {
+  describe('updateStatus', () => {
+    let req: Partial<Request>, res: Partial<Response>, userStub: SinonStub, mailerStub: SinonStub;
+
+    beforeEach(() => {
+      req = {
+        params: { userId: '123' },
+        body: { description: 'User violated terms of service' }
+      };
+      res = {
+        status: sinon.stub().returnsThis(),
+        json: sinon.stub()
+      };
+
+      userStub = sinon.stub(User, 'findByPk');
+    });
+
+    afterEach(() => {
+      sinon.restore();
+    });
+
+    it('should return 400 if description is missing for deactivation', async () => {
+      req.body!.description = '';
+      await adminController.updateStatus(req as Request, res as Response, 'inactive');
+
+      expect(res.status).to.have.been.calledWith(400);
+      expect(res.json).to.have.been.calledWith({ message: 'Description is required for deactivation' });
+    });
+
+    it('should return 404 if user is not found', async () => {
+      userStub.resolves(null);
+
+      await adminController.updateStatus(req as Request, res as Response, 'inactive');
+
+      expect(res.status).to.have.been.calledWith(404);
+      expect(res.json).to.have.been.calledWith({ message: 'User not found' });
+    });
+
+    it('should return 400 if user is already inactive', async () => {
+      userStub.resolves({ status: 'inactive' } as any);
+
+      await adminController.updateStatus(req as Request, res as Response, 'inactive');
+
+      expect(res.status).to.have.been.calledWith(400);
+      expect(res.json).to.have.been.calledWith({ message: 'User is already inactive' });
+    });
+
+    it('should update user status and send email for deactivation', async () => {
+      const user = { status: 'active', save: sinon.stub().resolves(), username: 'testuser', email: 'test@example.com' } as any;
+      userStub.resolves(user);
+
+      await adminController.updateStatus(req as Request, res as Response, 'inactive');
+
+      expect(user.status).to.equal('inactive');
+      expect(user.save).to.have.been.calledOnce;
+      expect(res.status).to.have.been.calledWith(200);
+      expect(res.json).to.have.been.calledWith({ message: 'User is now inactive, email sent successfully' });
+    });
+
+    // it('should handle email sending errors gracefully', async () => {
+    //   const user = { status: 'active', save: sinon.stub().resolves(), username: 'testuser', email: 'test@example.com' } as any;
+    //   userStub.resolves(user);
+
+    //   await adminController.updateStatus(req as Request, res as Response, 'inactive');
+
+    //    expect(res.status).to.have.been.calledWith(500);
+    // });
+  });
```

Comment:
If you are not using this test its better to remove it instead of commenting it.